core/trino-main/src/main/java/io/trino/event/QueryMonitor.java

```diff
@@ -249,6 +257,7 @@ private QueryMetadata createQueryMetadata(QueryInfo queryInfo)
                 queryInfo.getRoutines(),
                 queryInfo.getSelf(),
                 createTextQueryPlan(queryInfo),
+                createAnonymizedPlan(queryInfo),
```

Comment:
We already send the full plan with all details via `createTextQueryPlan` on above line in the query completion event.
QueryContext also has stuff like user and principal. So I don't think adding the anonymised plan is going to expose any new info to the query completion events.
If we must introduce a toggle for anonymisation of plan sent in query completion event, then it would make more sense to send the anonymised plan in place of the current full text query plan.

Reply:
Yeah - true. I did not know we have a plan in QCE already :)

Reply:
So why are adding new field for anonymized plan. It would make more sense (to me) to reuse the field we already have, and have a toggle which tell Trino, should normal or anonymized plan be recorded.

Reply:
Or, anonymization done in the event consumer.

Reply:
> Or, anonymization done in the event consumer.

The problem with anonymization in the event consumer is that the plan we get there is in text format and it's very hard to parse that plan to anonymize symbols, expressions or literals. Also, extracting any useful information from that plan is also not easy through code. That's why we want this anonymised plan in JSON format so that we could easily parse it later on.

Reply:
> why are adding new field for anonymized plan

It's possible that someone wants to record the full text plan and anonymised plan in different downstream systems to support different use cases. Currently we send the same event to all event listeners, so reusing the same field would take away some flexibility here.